metrics/src/main/scala/net/playq/metrics/base/MetricDef.scala

```diff
@@ -46,3 +49,27 @@ object MetricDef {
 
   def encodeAll(metrics: List[MetricDef]): String = metrics.map(encode).mkString("\n")
 }
+
+private[metrics] sealed trait LowPriorityInstances {
+  implicit def encoderFromCirce[R[_]](implicit @unused enc: _Encoder[R], F0: R[String]) = {
+    val F = F0.asInstanceOf[io.circe.Encoder[String]]
+    F.contramap[MetricDef](MetricDef.encode)
```

Comment:
This will cause the metric object to be encased in a wrapper string like: `"{ \"timer\": { ... } }"` not sure if worth it /cc @aliubymov-playq

Reply:
but that is a valid JSON string.
@neko-kai, you mean, that I need to use `derivation.deriveEncoder` here?

Reply:
@neko-kai Yup, totally agree. We should not use primitive codecs to create circe objects. 

Or we should do this: `io.circe.parser.parse(MetricDef.encode(metric))`. But that's a bad idea.
Better to define full codecs. 

(But for http purposes you still can send stringified by `MetricDef.encode` metrics.)

Reply:
@vpodlnk yeah, it's a valid JSON string, but we should create JSON object instead. Creating of JSON string wich represents a JSON object will require double decodings and could be mindblowing for users.

Reply:
@neko-kai, @aliubymov-playq  I've pushed a new version, I'm not sure that it is correct, cuz I can omit all these stuff required by `no-more-orphans` pattern. In case you have some free time, could you please verify whether on not I'm moving in the right direction?

Reply:
@vpodlnk Check this branch https://github.com/PlayQ/d4s/pull/61 - if you remove `.asInstanceOf[R[_]]` you'll get compile errors:

```
Error:(11, 45) Symbol 'type io.circe.Encoder.AsObject' is missing from the classpath.
This symbol is required by 'method net.playq.metrics.base.LowPriorityInstances.encoderFromCirce'.
Make sure that type AsObject is in your classpath and check for conflicting dependencies with `-Ylog-classpath`.
A full rebuild may help if 'LowPriorityInstances.class' was compiled against an incompatible version of io.circe.Encoder.
    assert(summonOrNull[D4SAttributeEncoder[MetricDef]] == null)
```

Note that the test is in `d4s` module, not in `metrics` module. because optional dependencies from the main scope are also on classpath in the test scope of metrics, which is why you didn't need the tricks there. (This is in Intellij, actually I'm not sure whether that applies to sbt - maybe sbt test scope will _not_ include the optional libraries)

Reply:
@neko-kai, thanks for review and explanation!